tests/utils/creditNotePreview.js

```diff
@@ -0,0 +1,130 @@
+        initialAmount: random.randomInt(1, 9999),
+        interval: _choiceBetween(choice(["month", "year"]))
+    })
+}
+
+function generatePricePreview(){
+    return new starkinfra.CreditNotePreview({
+        taxId: TaxIdGenerator.taxId(),
+        type: "price",
+        nominalAmount: random.randomInt(1, 100000),
+        rebateAmount: _choiceBetween(random.randomInt(1, 1000)),
+        nominalInterest: random.randomFloat(0, 4.99),
+        scheduled: random.futureDate(days=random.randomInt(11, 20)),
+        initialDue: random.futureDate(days=random.randomInt(30, 40)),
+        initialAmount: random.randomInt(1, 9999),
+        interval: _choiceBetween(choice(["month", "year"]))
+    })
+}
+
+function generateAmericanPreview(){
+    return new starkinfra.CreditNotePreview({
+        taxId: TaxIdGenerator.taxId(),
+        type: "american",
+        nominalAmount: random.randomInt(1, 100000),
+        rebateAmount: _choiceBetween(random.randomInt(1, 1000)),
+        nominalInterest: random.randomFloat(0, 4.99),
+        scheduled: random.futureDate(days=random.randomInt(11, 20)),
+        initialDue: random.futureDate(days=random.randomInt(30, 40)),
+        count: random.randomInt(1, 100),
+        interval: _choiceBetween(choice(["month", "year"]))
+    })
+}
+
+function generateBulletPreview(){
+    return new starkinfra.CreditNotePreview({
+        taxId: TaxIdGenerator.taxId(),
+        type: "bullet",
+        nominalAmount: random.randomInt(1, 100000),
+        rebateAmount: _choiceBetween(random.randomInt(1, 1000)),
+        nominalInterest: random.randomFloat(0.1, 4.99),
+        scheduled: random.futureDate(days=random.randomInt(11, 20)),
+        initialDue: random.futureDate(days=random.randomInt(30, 40))
+    })
+}
+
+function generateCustomPreview(){
+    var amount = random.randomInt(1, 100000)
+    return new starkinfra.CreditNotePreview({
+        taxId: TaxIdGenerator.taxId(),
+        type: "custom",
+        nominalAmount: amount,
+        scheduled: random.futureDate(days=random.randomInt(11, 20)),
+        rebateAmount: _choiceBetween(random.randomInt(1, 1000)),
+        invoices: generateInvoicePreview(n=random.randomInt(1, 10), amount)
+    })
+}
+
+function generateRandomPreview(){
+    return choice([
+        generateSacPreview,
+        generatePricePreview,
+        generateAmericanPreview,
+        generateBulletPreview,
+        generateCustomPreview
+    ]).call()
+}
+
+const generatorsByType = {
+    'sac': generateSacPreview,
+    'price': generatePricePreview,
+    'american': generateAmericanPreview,
+    'bullet': generateBulletPreview,
+    'custom': generateCustomPreview,
+}
+
+function getPreviewExample(type){
+    return type ? generatorsByType[type].call() : generateRandomPreview()
+}
+    
+
+
+function getPreviewExamples(n=10, type){
+    return type ? [...Array(n)].map(() => getPreviewExample(type)) :
+                  [...Array(n)].map(() => getPreviewExample())
```

Comment:
Pode deixar esse ternário inline mesmo. Se for quebrar a linha, faz usando o if explícito